doc/howtos/using_ubuntu_as_sos.rst

```diff
@@ -0,0 +1,184 @@
+
+# mkdir ~/kernel-build
+# cd ~/kernel-build
+# wget  https://download.clearlinux.org/releases/current/clear/x86_64/os/Packages/linux-pk414-sos-4.14.41-39.x86_64.rpm
+# sudo apt-get install rpm2cpio
+# rpm2cpio linux-pk414-sos-4.14.41-39.x86_64.rpm | cpio -idmv 
+# sudo cp -r ~/kernel-build/usr/lib/modules/4.14.41-39.pk414-sos/ /lib/modules/
+
+Step #8, Deploying the hypervisor
+Mount the EFI partition and verify you have the following files, then add the ACRN hypervisor and Clear Linux kernel to the EFI 
+Partition, 
+# umount /boot/efi 
+# lsblk
+# mount /dev/sda1 /mnt
+# ls /mnt/EFI/ubuntu
+  fw  fwupx64.efi  grub.cfg  grubx64.efi  MokManager.efi  shimx64.efi
+
+Copy acrn.efi  to /mnt/EFI/acrn/ 
+# sudo mkdir /mnt/EFI/acrn/
+# sudo cp acrn-hypervisor/build/acrn.efi /mnt/EFI/acrn	
+
+Step #9, Configure the EFI firmware to boot the ACRN hypervisor by default
+# sudo -s
+# efibootmgr -c -l "\EFI\acrn\acrn.efi" -d /dev/sda -p 1 -L "ACRN Hypervisor                                                                                                    NUC" -u "bootloader=\EFI\ubuntu\grubx64.efi"
+Then you can see the result of boot order. Also you can check the boot order with below command:
+# efibootmgr -v            	
+or setting boot order with this command to make sure the acrn.efi is the first boot option:
+# efibootmgr -o x, x       	
+Note: by default, the “ACRN Hypervisor” you just added should be the first one to boot. But please make sure it is, or select it 
+manually after press “F10” when system booting up.   
+
+Step #10, Deploy the SOS kernel
+# sudo cp ~/kernel-build/usr/lib/kernel/org.clearlinux.pk414-sos.4.14.41-39 /boot/acrn/
+
+Step #11, Customize the boot of the grub entry 
+Use grub custom file (/etc/grub.d/40_custom) to create a new grub entry
+Add the following to the end of grub file
+menuentry 'ACRN ubuntu SOS' {
+        recordfail
+        load_video
+        insmod gzio
+        insmod part_gpt
+        insmod ext4
+        linux /boot/acrn/org.clearlinux.pk414-sos.4.14.41-39 pci_devices_ignore=(0:18:1) maxcpus=1 console=tty0 console=ttyS0 
+        i915.nuclear_pageflip=1 root=PARTUUID=<ID of rootfs partition> rw rootwait ignore_loglevel no_timer_check consoleblank=0 
+        i915.tsd_init=7 i915.tsd_delay=2000 i915.avail_planes_per_pipe=0x00000F i915.domain_plane_owners=0x011111110000 
+        i915.enable_guc_loading=0 i915.enable_guc_submission=0 i915.enable_preemption=1 i915.context_priority_mode=2 
+        i915.enable_gvt=1 hvlog=2M@0x1FE00000
+}
+Note: here you have to use PARTUUID, UUID doesn’t work.  
+
+Step #12,  Update grub
+# sudo update-grub
+At this moment, you need to modify /boot/grub/grub.cfg file manually to enable the timeout so that the system has opportunity to show 
+you grub menu to select. Or you need to set the “ACRN ubuntu SOS” as the default boot option. Without enabling  timeout, the system 
+will not show grub menu for people to select and boot into incorrect kernel.
+
+Enable timeout:
```

Comment:
This notation might be confusing (with the leading ``+`` and ``-`` signs).  Better would be to just tell them what it should look like after the changes are made.

Reply:
Enabling the time-out and changing the default boot target should always be done by editing `/etc/default/grub` (and **then** running `sudo update-grub`). If something (bug or something else) is forcing us to edit the `/boot/grub.cfg` directly, we should explain this here.

Reply:
no solution so far, we don't need to modify it manually until  having good tool for that

Reply:
> no solution so far, we don't need to modify it manually until having good tool for that

Are you saying it's not possible to make these modifications in `/etc/default/grub` directly and then run `sudo update-grub`?